indexer/feature_data.cpp

```diff
@@ -500,10 +500,17 @@ bool FeatureParams::FinishAddingTypes()
   sort(newTypes.begin(), newTypes.end());
   newTypes.erase(unique(newTypes.begin(), newTypes.end()), newTypes.end());
 
-  m_types.swap(newTypes);
+  if (newTypes.size() > kMaxTypesCount)
+  {
+    // Put common types to the end to leave the most important types.
+    static UselessTypesChecker checker;
+    UNUSED_VALUE(base::RemoveIfKeepValid(newTypes.begin(), newTypes.end(),
```

Comment:
nit: Can we just pass checker as 3rd param?

Reply:
done